checks/prometheus_check.py

```diff
@@ -0,0 +1,339 @@
+# (C) Datadog, Inc. 2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import re
+import requests
+from google.protobuf.internal.decoder import _DecodeVarint32  # pylint: disable=E0611,E0401
+from checks import AgentCheck
+from utils.prometheus import metrics_pb2
+
+# Prometheus check is a mother class providing a structure and some helpers
+# to collect metrics, events and service checks exposed via Prometheus.
+#
+# It must be noted that if the check implementing this class is not officially
+# supported
+# its metrics will count as cutom metrics and WILL impact billing.
+#
+# Minimal config for checks based on this class include:
+#   - implementing the check method
+#   - overriding self.NAMESPACE
+#   - overriding self.metric_to_gauge
+#     AND/OR
+#   - create method named after the prometheus metric they will handle (see self.prometheus_metric_name)
```

Comment:
we also removed that example, either re-introduce it as it was written in https://github.com/DataDog/dd-agent/commit/9be9af8bd49157c2985b29359ebf9c0707906383#diff-b97e21cc8c9598269fd1833ccdd51ae3R154 or as a commented-out method -- it could be something like 
```python
def prometheus_metric_name(self, message, **kwargs):
    if message.type < len(self.METRIC_TYPES):
        for metric in message.metric:
            val = getattr(metric, self.METRIC_TYPES[message.type]).value
            tags = ['{}:{}'.format(label.name, label.value) for label in metric.label]
            self.gauge("datadog.metric.name", val, tags)
```
Or just remove the mention to it.